src/serializers/services-serializer.js

```diff
@@ -0,0 +1,92 @@
+import { Serializer as JsonApiSerializer } from 'jsonapi-serializer';
+import _ from 'lodash';
+
+import { serializerOptions } from 'utils/jsonapi';
+import { openapi } from 'utils/load-openapi';
+import { paginate } from 'utils/paginator';
+import { apiBaseUrl, resourcePathLink, paramsLink } from 'utils/uri-builder';
+
+const serviceResourceProp = openapi.components.schemas.ServiceResource.properties;
+const serviceResourceType = serviceResourceProp.type.enum[0];
+const serviceResourceKeys = _.keys(serviceResourceProp.attributes.properties);
+const serviceResourcePath = 'services';
+const serviceResourceUrl = resourcePathLink(apiBaseUrl, serviceResourcePath);
+
+/**
+ * Serialize serviceResources to JSON API
+ *
+ * @param {object[]} rawServices Raw data rows from data source
+ * @param {object} req Express request object
+ * @returns {object} Serialized serviceResources object
+ */
+const serializeServices = (rawServices, req) => {
+  const { query } = req;
+
+  // Add pagination links and meta information to options if pagination is enabled
+  const pageQuery = {
+    size: query['page[size]'],
+    number: query['page[number]'],
+  };
+
+  const pagination = paginate(rawServices, pageQuery);
+  pagination.totalResults = rawServices.length;
+  rawServices = pagination.paginatedRows;
+
+  // TODO use req.path
+  const topLevelSelfLink = paramsLink(serviceResourceUrl, query);
+  const serializerArgs = {
+    identifierField: 'id',
+    resourceKeys: serviceResourceKeys,
+    pagination,
+    resourcePath: serviceResourcePath,
+    topLevelSelfLink,
+    query,
+    enableDataLinks: true,
+  };
+
+  // flatten attributes object in rawServices for serializer
+  const newRawServices = [];
+  _.forEach(rawServices, (rawService) => {
+    newRawServices.push({
+      ...{ id: rawService.id, type: rawService.type },
```

Comment:
I could be wrong here but I doesn't `...{}` cancel each other out here? I think you can do 
```
    newRawServices.push({
      id: rawService.id,
      type: rawService.type,
```

Reply:
you're right!